### PR TITLE
New API nested filters.

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description

Add schema generation step for nested filters on the new API

### CODE REVIEW

File: packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts [hunk 1]
```diff
@@ -49,16 +56,32 @@ export class TopLevelFilterSchemaTypes extends FilterSchemaTypes<ConcreteEntity>
...
                     OR: itc.NonNull.List,
                     NOT: itc,
                     ...this.convertAttributesToFilters([...this.entity.attributes.values()]),
+                    ...this.convertRelationshipsToFilters([...this.entity.relationships.values()]),
                 },
             };
         });
     }
+
+    private convertRelationshipsToFilters(relationships: Relationship[]): Record<string, InputTypeComposer> {
```
Comment (angrykoala) [inline packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts]: Maybe `createRelationshipFilterTypes` is a better name?

File: packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts [hunk 2]
```diff
@@ -49,16 +56,32 @@ export class TopLevelFilterSchemaTypes extends FilterSchemaTypes<ConcreteEntity>
...
+        const fields: ([string, InputTypeComposer | GraphQLScalarType] | [])[] = filterTruthy(
```
Comment (angrykoala) [inline packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts]: Maybe it is more readable with Array?
```suggestion
        const fields: Array<[string, InputTypeComposer | GraphQLScalarType] | []> = filterTruthy(
```

Comment (angrykoala) [inline packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts]: Also, why filterTruthy?

it looks like this map always return a value

**Review by angrykoala — APPROVED** [review]

File: packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts (hunk 1 above)
Comment (MacondoExpress) [reply packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts]: Sounds good! I'll be changing `convertAttributeToFilters` to `createPropertyFilters` as well in this case!

File: packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts (hunk 2 above)
Comment (MacondoExpress) [reply packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts]: No problem!

Comment (MacondoExpress) [reply packages/graphql/src/api-v6/schema-generation/schema-types/filter-schema-types/TopLevelFilterSchemaTypes.ts]: Good catch I  changed it at the last minute but I forgot to remove `filterTruthy`

**Pull request merged**